crates/core/src/db/datastore/system_tables/test_one_point_oh_layouts.rs

```diff
@@ -0,0 +1,423 @@
+    Layout,
+    PrimitiveType::*,
+    ProductTypeElementLayout, ProductTypeLayout, SumTypeLayout, SumTypeVariantLayout,
+    VarLenType::*,
+};
+
+use super::{
+    StClientRow, StColumnRow, StConstraintRow, StIndexRow, StModuleRow, StScheduledRow, StSequenceRow, StTableRow,
+    StVarRow,
+};
+
+pub(super) trait HasOnePointOhLayout: SpacetimeType {
+    /// Get the original layout of the type as of 1.0.
+    /// Note: if the logic for computing layouts ever changes, you'll need to think VERY CAREFULLY
+    /// about how to update this function.
+    fn get_original_layout() -> AlgebraicTypeLayout;
+}
+
+macro_rules! boxed {
+    ($($var:expr),* $(,)?) => {
+        vec![$($var),*].into()
+    };
+}
+
+fn col_list_layout() -> AlgebraicTypeLayout {
+    VarLen(Array(Box::new(AlgebraicType::array(AlgebraicType::U16))))
+}
+
+fn bytes_layout() -> AlgebraicTypeLayout {
+    VarLen(Array(Box::new(AlgebraicType::array(AlgebraicType::U8))))
+}
+
+impl HasOnePointOhLayout for StTableRow {
+    fn get_original_layout() -> AlgebraicTypeLayout {
+        Product(ProductTypeLayout {
+            layout: Layout { size: 24, align: 4 },
+            elements: boxed![
+                ProductTypeElementLayout {
+                    offset: 0,
+                    ty: Primitive(U32),
+                    name: Some("table_id".into()),
+                },
+                ProductTypeElementLayout {
+                    offset: 4,
+                    ty: VarLen(String),
+                    name: Some("table_name".into()),
+                },
+                ProductTypeElementLayout {
+                    offset: 8,
+                    ty: VarLen(String),
+                    name: Some("table_type".into()),
+                },
+                ProductTypeElementLayout {
+                    offset: 12,
+                    ty: VarLen(String),
+                    name: Some("table_access".into()),
+                },
+                ProductTypeElementLayout {
+                    offset: 16,
+                    ty: Sum(SumTypeLayout {
+                        layout: Layout { size: 6, align: 2 },
+                        variants: boxed![
+                            SumTypeVariantLayout {
+                                ty: col_list_layout(),
+                                name: Some("some".into()),
+                            },
+                            SumTypeVariantLayout {
+                                ty: Product(ProductTypeLayout {
+                                    layout: Layout { size: 0, align: 1 },
+                                    elements: Box::new([]),
+                                }),
+                                name: Some("none".into()),
+                            },
+                        ],
+                        payload_offset: 2,
+                    }),
+                    name: Some("table_primary_key".into()),
+                },
+            ],
+        })
+    }
+}
```

Comment:
Similar calls of this could benefit by a helper like:

```rust
impl HasOnePointOhLayout for StTableRow {
    fn get_original_layout() -> AlgebraicTypeLayout {
        Product(ProductTypeLayout {
            layout: Layout { size: 24, align: 4 },
            elements: boxed![
                element_layout(0, Primitive(U32), "table_id"),
                element_layout(4, VarLen(String), "table_name"),
                element_layout(8, VarLen(String), "table_type"),
                element_layout(12, VarLen(String), "table_access"),
                element_layout(16, ..., "table_primary_key"),
            ],
        })
    }
}
```